tools/static_word_checks.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import os
+import sys
+
+
+class StaticCheck:
+    """Run simple checks on the entire document or specific lines."""
+
+    def __init__(self, replace: bool):
+        self.full_file_checks = [
+            # add more checks to the list as they are added
+        ]
+        self.line_checks = [
+            self.check_for_noreturn,
+            self.check_for_double_r,
+            self.check_for_double_u,
+            # add more checks to the list as they are added
+        ]
+        self.replace = replace
+
+    def check_for_noreturn(
+        self, filename: str, lineno: str, content: str
+    ) -> bool:
+        if "noreturn" in content.lower():
+            print(
+                "{filename}: Line {line_number} has phrase noreturn, "
+                "please edit and remove this.".format(
+                    filename=filename, line_number=lineno
+                )
+            )
+            return False
+        return True
+
+    def check_for_double_r(
```

Comment:
This method on the other method _should_ be the same save for `/r` compared to`/u/. Can we make a single method that takes `r` or `u` as an argument?

Also, consider using a regular expression for the replacement:

```python
re.sub("(^|\s)/r/", r"\1r/", "foobar /r/foo")
```

Reply:
I modified your regex to `(^|\s)/(u|r)/` so both `/r/` and `/u/` are replaced, and a capture group is used for that.